core/common/src/main/java/alluxio/conf/AlluxioProperties.java

```diff
@@ -189,6 +201,15 @@ public void forEach(BiConsumer<? super PropertyKey, ? super String> action) {
     }
   }
 
+  /**
+   * Makes a copy of the backing properties and returns them in a new object.
+   *
+   * @return a copy of the current properties
+   */
+  public AlluxioProperties copy() {
+    return new AlluxioProperties(this);
```

Comment:
Does this need to be threadsafe?

Reply:
The class is annotated as `NotThreadSafe`. If we want it to be thread safe we should analyze all methods in this class and remove the annotation.

Otherwise- the AlluxioProperties object is never used used outside of the AlluxioConfiguration objects. The `AlluxioConfiguration` interface makes the the configuration objects immutable - which means to modify them you'd need to copy the properties anyways which will only happen in one thread.

Reply:
Got it, I was worried about the copy not getting a consistent state since there are two `putAll` calls which are not guaranteed to work if the backing map is being modified.